### DISCUSSION TITLE
Issue - Failed to enable ssl with JKS

### DISCUSSION THREAD

**Discussion by eronrunner** [apache/pulsar#17065] [Q&A]
I install pulsar by helm chart and try to add the properties

  brokerServicePortTls: "{{.Values.broker.ports.pulsarssl}}"
  webServicePortTls: "{{.Values.broker.ports.https}}"
  tlsEnabled: "true"
  tlsCertificateFilePath: "/pulsar/certs/manager.crt"
  tlsKeyFilePath: "/pulsar/certs/manager.key"
  tlsTrustCertsFilePath: "/pulsar/certs/ca.crt"
  tlsProtocols: "TLSv1.2,TLSv1.1"

  tlsEnabledWithKeyStore: "true"
  requireTrustedClientCertOnConnect: "true"
 
  tlsKeyStoreType: JKS
  tlsKeyStore: "{{.Values.tlsKeyStore.tlsKeyStore}}"
  tlsKeyStorePassword: "{{.Values.tlsKeyStore.tlsKeyStorePassword}}"
 
  tlsTrustStoreType: JKS
  tlsTrustStore: "{{.Values.tlsKeyStore.tlsTrustStore}}"
  tlsTrustStorePassword: "{{.Values.tlsKeyStore.tlsTrustStorePassword}}"
 
  brokerClientTlsEnabled: "{{.Values.tlsKeyStore.brokerClientTlsEnabled}}"
  brokerClientTlsEnabledWithKeyStore: "{{.Values.tlsKeyStore.brokerClientTlsEnabledWithKeyStore}}"
  brokerClientTlsTrustStoreType: JKS
  brokerClientTlsTrustStore: "{{.Values.tlsKeyStore.brokerClientTlsTrustStore}}"
  brokerClientTlsTrustStorePassword: "{{.Values.tlsKeyStore.brokerClientTlsTrustStorePassword}}"
![image](https://user-images.githubusercontent.com/35225206/184102521-68884166-b053-4ada-b4f4-a39bdbb26b55.png)

![image](https://user-images.githubusercontent.com/35225206/184097308-9fcc8dec-c289-432f-91ea-5a442f597b23.png)
![image](https://user-images.githubusercontent.com/35225206/184097915-b1c54f02-e378-4c45-8e8d-5e27800bafdb.png)
I also tried some ways by disabling  the HTTP port and non-ssl service port, but got the same things.
Please help!
Thank you

**Comment by asafm** [comment]
@hangc0276 ?

**Reply by nodece** [reply]
It looks related to DNS.

**Comment by tisonkun** [comment]
@eronrunner can you provide which version (os & pulsar) in used?

OK. It seems Pulsar version=2.9.2. If it's on MacOS, DNS resolution can meet issue described in https://github.com/apache/pulsar/discussions/17067.

**Comment by michaeljmarshall** [comment]
Note that the `Failed to resolve network location` log only indicates that you have not configured racks for your bookies. It is not preventing TLS from working, and it is not attempting to do any DNS resolution, assuming you're using the broker.conf defaults.

I think we'll likely need more of the logs to answer what is going wrong. Specifically, we'll need to see the full stack trace from the client, and we'll also need to see the full stack trace from the broker.

Note that the final error is coming from this block: https://github.com/apache/pulsar/blob/a3f52891593e093c27b583094a1fbfd09bbbae1a/pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/http/AsyncHttpConnector.java#L263-L266.
